src/Microsoft.VisualBasic.Core/tests/FileSystemTests.cs

```diff
@@ -0,0 +1,731 @@
+            if (PlatformDetection.IsWindows)
+            {
+                FileSystem.Rename(sourceName, destName);
+                Assert.False(System.IO.File.Exists(sourceName));
+                Assert.True(System.IO.File.Exists(destName));
+                Assert.Equal("abc", System.IO.File.ReadAllText(destName));
+            }
+            else
+            {
+                AssertThrows<PlatformNotSupportedException>(() => FileSystem.Rename(sourceName, destName));
+                Assert.True(System.IO.File.Exists(sourceName));
+                Assert.False(System.IO.File.Exists(destName));
+                Assert.Equal("abc", System.IO.File.ReadAllText(sourceName));
+            }
+
+            // Rename to an existing name.
+            sourceName = GetTestFilePath();
+            System.IO.File.WriteAllText(sourceName, "def");
+            destName = GetTestFilePath();
+            System.IO.File.WriteAllText(destName, "123");
+            if (PlatformDetection.IsWindows)
+            {
+                AssertThrows<System.IO.IOException>(() => FileSystem.Rename(sourceName, destName));
+            }
+            else
+            {
+                AssertThrows<PlatformNotSupportedException>(() => FileSystem.Rename(sourceName, destName));
+            }
+            Assert.True(System.IO.File.Exists(sourceName));
+            Assert.True(System.IO.File.Exists(destName));
+            Assert.Equal("def", System.IO.File.ReadAllText(sourceName));
+            Assert.Equal("123", System.IO.File.ReadAllText(destName));
+        }
+
+        // Not tested:
+        //   public static void Reset() { }
+        //   public static void Seek(int FileNumber, long Position) { }
+        //   public static long Seek(int FileNumber) { throw null; }
+        //   public static void SetAttr(string PathName, FileAttribute Attributes) { }
+        //   public static SpcInfo SPC(short Count) { throw null; }
+        //   public static TabInfo TAB() { throw null; }
+        //   public static TabInfo TAB(short Column) { throw null; }
+        //   public static void WriteLine(int FileNumber, params object[] Output) { }
+
+        [Fact]
+        public void Write_ArgumentException()
+        {
+            int fileNumber = FileSystem.FreeFile();
+            var fileName = GetTestFilePath();
+            FileSystem.FileOpen(fileNumber, fileName, OpenMode.Output);
+            AssertThrows<ArgumentException>(() => FileSystem.Write(fileNumber, new object()));
+            FileSystem.FileClose(fileNumber);
+        }
+
+        // We cannot use XUnit.Assert.Throws<T>() for lambdas that rely on AssemblyData (such as
+        // file numbers) because AssemblyData instances are associated with the calling assembly, and
+        // in RELEASE builds, the calling assembly of a lambda invoked from XUnit.Assert.Throws<T>()
+        // is corelib rather than this assembly, so file numbers created outside the lambda will be invalid.
```

Comment:
@cston, I just tried replacing this with Assert.Throws, and all tests passed.  Can you elaborate on what's broken without this and why it's needed?